velox/common/base/SafeInt64MaxComparison.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <limits>
+
+#include "folly/CPortability.h"
+
+namespace facebook::velox {
+
+inline constexpr long kLongMax = std::numeric_limits<int64_t>::max();
+
+// On some compilers if we cast kLongMax to a double, we can get a number larger
+// than 'kLongMax'. This will allow 'val' values > 'kLongMax'. The workaround
+// here is to use uint64_t to represent ('kLongMax' + 1), which can be
+// represented exactly as double. We then check if the difference with 'val'
+// >= 1.
+FOLLY_ALWAYS_INLINE bool safeLessThanOrEqualToInt64Max(double val) {
+  return (static_cast<uint64_t>(kLongMax) + 1) - val >= 1;
```

Comment:
Do we expect the compiler to add an implicit cast of `(static_cast<uint64_t>(kLongMax) + 1)` to double? If so, would it be clearer to define a constant of type double?